### ISSUE TITLE
Move IfcProducts within file

### ISSUE THREAD

**Issue by bimworld** [IfcOpenShell/IfcOpenShell#2252]
Does the C++ version provide some functions to move an IfcProduct (and the linked subproducts) from one hierarchical element (e.g. storey or space) to another keeping the actual placement and rotation within the building? I want to move the IfcProducts linked to an IfcSpace to the storey the space is in.

**Comment by aothms** [comment]
No, no direct API, but the changing the relative placement without changing the final position, is a matter of figuring out the correct occt operations wrt `gp_Trsf` `Inverted()` `Multiply()`/`PreMultiply()` Conceptually, something like: IfcProduct.ObjectPlacement.RelativePlacement := convert(IfcProduct.ObjectPlacement) * convert(IfcSpace.ObjectPlacement).Inverted()
